ddtrace/debugging/_origin/span.py

```diff
@@ -152,36 +154,39 @@ def __enter__(self):
             s.set_tag_str("_dd.code_origin.frames.0.type", location.module)
             s.set_tag_str("_dd.code_origin.frames.0.method", location.name)
 
-        # TODO[gab]: This will be enabled as part of the live debugger/distributed debugging
-        # if ld_config.enabled:
-        #     # Create a snapshot
-        #     snapshot = Snapshot(
-        #         probe=location.probe,
-        #         frame=self.__frame__,
-        #         thread=current_thread(),
-        #         trace_context=root,
-        #     )
+        # Check if we have any level 2 debugging sessions running for the
+        # current trace
+        if any(s.level >= 2 for s in Session.from_trace()):
```

Comment:
I think this check should only happen on exit. as the trigger probe can mark the trace as captured in the middle of the span.